ipmininet/router/config/bgp.py

```diff
@@ -175,7 +313,7 @@ def _find_peer_address(base, peer, v6=False):
         # Explore all interfaces in base ASN recursively, until we find one
         # connected to the peer
         while to_visit:
-            i = to_visit.pop()
+            i = to_visit.pop(0)
```

Comment:
It was to force a bfs for the exploration of neighbors because, before, as pop removes the last item, to_visit was used as a stack and it was then a dfs used for the neighbor exploration

Reply:
Well, it works in both cases, so you shouldn't lengthen your PR to change unrelated things that already work. Please remove that from your PR.

Reply:
Antoine told me he added it because there were conflicts in bgp peering connections. A router tried to communicate via eth0 to its neighbor but via the dfs, the neighbor found its path towards the router via eth1 and the bgp session was failing

Reply:
Ok, then, this change should appear in a commit alone with an explicit commit message detailing the issue and why it must be changed.
This is annoying to do but very valuable for future developers :)